File: DevExpress-Demo/CS/EditorsDemo.Wpf/Helpers/BarCodeSymbologyTemplateSelector.cs
using DevExpress.Xpf.Editors;
using System.Windows;
using System.Windows.Controls;

namespace EditorsDemo {
    public class BarCodeSymbologyTemplateSelector : DataTemplateSelector {
        public override System.Windows.DataTemplate SelectTemplate(object item, System.Windows.DependencyObject container) {
            FrameworkElement element = container as FrameworkElement;
            if(element == null || item == null || !(item is BarCodeStyleSettings))
                return base.SelectTemplate(item, container);
            return (DataTemplate)element.FindResource(((BarCodeStyleSettings)item).GeneratorBase.Name);
        }
    }
}

File: DevExpress-Demo/VB/EditorsDemo.Wpf/Helpers/BarCodeSymbologyTemplateSelector.vb
Imports DevExpress.Xpf.Editors
Imports System.Windows
Imports System.Windows.Controls

Namespace EditorsDemo

    Public Class BarCodeSymbologyTemplateSelector
        Inherits DataTemplateSelector

        Public Overrides Function SelectTemplate(ByVal item As Object, ByVal container As DependencyObject) As DataTemplate
            Dim element As FrameworkElement = TryCast(container, FrameworkElement)
            If element Is Nothing OrElse item Is Nothing OrElse Not(TypeOf item Is BarCodeStyleSettings) Then Return MyBase.SelectTemplate(item, container)
            Return CType(element.FindResource(CType(item, BarCodeStyleSettings).GeneratorBase.Name), DataTemplate)
        End Function
    End Class
End Namespace
